evap/staff/importers.py

```diff
@@ -653,21 +680,29 @@ def process_contributors(self, evaluation, test_run, user_list):
         already_related = [contribution.contributor for contribution in already_related_contributions]
         if already_related:
             msg = format_html(_("The following {} users are already contributing to evaluation {}:"), len(already_related), evaluation.full_name)
+            msg = format_html(ngettext('The following user is already contributing to evaluation %(name)s:', 'The following %(users)d users are already contributing to evaluation %(name)s:', len(already_related),) % {'users': len(already_related),'name': evaluation.full_name})
             msg += create_user_list_html_string_for_message(already_related)
             self.warnings[ImporterWarning.GENERAL].append(msg)
 
         # since the user profiles are not necessarily saved to the database, they are not guaranteed to have a pk yet which
         # makes anything relying on hashes unusable here (for a faster list difference)
         users_to_add = [user for user in user_list if user not in already_related]
 
-        if not test_run:
-            for user in users_to_add:
-                order = Contribution.objects.filter(evaluation=evaluation).count()
-                Contribution.objects.create(evaluation=evaluation, contributor=user, order=order)
-            msg = format_html(_("{} contributors added to the evaluation {}:"), len(users_to_add), evaluation.full_name)
+        if len(users_to_add) == 0:
+            if not test_run:
+                evaluation.participants.add(*users_to_add)
+                msg = format_html(_("Zero contributors added to the evaluation {}"), evaluation.full_name)
+            else:
+                msg = format_html(_("Zero contributors would be added to the evaluation {}"), evaluation.full_name)
         else:
-            msg = format_html(_("{} contributors would be added to the evaluation {}:"), len(users_to_add), evaluation.full_name)
-        msg += create_user_list_html_string_for_message(users_to_add)
+            if not test_run:
+                for user in users_to_add:
+                    order = Contribution.objects.filter(evaluation=evaluation).count()
+                    Contribution.objects.create(evaluation=evaluation, contributor=user, order=order)
+                msg = format_html(ngettext('One contributor added to the evaluation %(name)s:', '%(user)s contributors added to the evaluation %(name)s:', len(users_to_add),) % {'user': len(users_to_add), 'name': evaluation.full_name})
+            else:
+                msg = format_html(ngettext('One contributor would be added to the evaluation %(name)s:', '%(user)s contributors would be added to the evaluation %(name)s:', len(users_to_add),) % {'user': len(users_to_add), 'name': evaluation.full_name})
+            msg += create_user_list_html_string_for_message(users_to_add)
```

Comment:
I feel like you can reduce the intersection between the branches a bit, maybe start with `msg = ""` before this hunk, then conditionally add the count and the words `"contributor"` or `"contributors"`, then check for `test_run` (and append `"would be"`), and so on. I don't know if it would play out as nicely as I am imagining it right now

Reply:
keeping translations in mind, it is better to create complete sentences and not to add words to the middle of a sentence